x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -807,38 +808,66 @@ public void updateConnectorLastSyncStats(UpdateConnectorLastSyncStatsAction.Requ
      *
      * @param request  The request for updating the connector's is_native property.
      * @param listener The listener for handling responses, including successful updates or errors.
      */
     public void updateConnectorNative(UpdateConnectorNativeAction.Request request, ActionListener<UpdateResponse> listener) {
         try {
             String connectorId = request.getConnectorId();
+            boolean isNative = request.isNative();
 
-            final UpdateRequest updateRequest = new UpdateRequest(CONNECTOR_INDEX_NAME, connectorId).doc(
-                new IndexRequest(CONNECTOR_INDEX_NAME).opType(DocWriteRequest.OpType.INDEX)
-                    .id(connectorId)
-                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-                    .source(
-                        Map.of(
-                            Connector.IS_NATIVE_FIELD.getPreferredName(),
-                            request.isNative(),
-                            Connector.STATUS_FIELD.getPreferredName(),
-                            ConnectorStatus.CONFIGURED.toString()
-                        )
-                    )
+            getConnector(connectorId, listener.delegateFailure((l, connector) -> {
 
-            );
-            client.update(updateRequest, new DelegatingIndexNotFoundActionListener<>(connectorId, listener, (l, updateResponse) -> {
-                if (updateResponse.getResult() == UpdateResponse.Result.NOT_FOUND) {
-                    l.onFailure(new ResourceNotFoundException(connectorNotFoundErrorMsg(connectorId)));
+                String indexName = getConnectorIndexNameFromSearchResult(connector);
+
+                // Ensure attached content index is prefixed correctly
+                if (isNative && indexName != null && isValidManagedConnectorIndexName(indexName) == false) {
+                    l.onFailure(
+                        new ElasticsearchStatusException(
+                            "The index name ["
+                                + indexName
+                                + "] attached to the connector ["
+                                + connectorId
+                                + "] must start with the required prefix: ["
+                                + MANAGED_CONNECTOR_INDEX_PREFIX
+                                + "] to be Elastic-managed. Please update the attached index first to comply with this requirement.",
+                            RestStatus.BAD_REQUEST
+                        )
+                    );
                     return;
                 }
-                l.onResponse(updateResponse);
-            }));
 
+                ConnectorStatus status = getConnectorStatusFromSearchResult(connector);
+
+                // If connector was connected already, change its status to CONFIGURED as we need to re-connect
+                if (status == ConnectorStatus.CONNECTED) {
```

Comment:
I realised we would always set `CONFIGURED` once changing the `is_native` flag - it's likely wrong, we got here because I copied logic from Kibana initially https://github.com/elastic/kibana/pull/175536/files